### PR TITLE
Migrate to python3.10 and apply patches.

### PR DESCRIPTION
**Pull request by erykoff**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

This applies the proposed upstream patches from https://github.com/docker/docker-py/pull/2857

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,9 +7,12 @@ package:
 source:
   url: https://github.com/docker/docker-py/archive/{{ version }}.tar.gz
   sha256: 0035772df94a86e8db02aac363393755584383aacf50d37b0d55ea817149e1f4
+  patches:
+    - 0001-loosen-pywin32-pin.patch
+    - 0002-update-test-skips.patch
```
Comment (ocefpaf) [inline recipe/meta.yaml:12]: Are these upstream or upstream-able? If the former please add a link to the commit or PR that added them. If the latter than consider sending them upstream.

File: recipe/0001-loosen-pywin32-pin.patch [hunk 2]
```diff
@@ -1,3 +1,4 @@
+Adapted from https://github.com/docker/docker-py/pull/2857/commits/2e8975698eb3bc36cb037e601ee798eace9bcc53
```
Comment (ocefpaf) [inline recipe/0001-loosen-pywin32-pin.patch:1]: Please add those to the recipe as a comment at the top of the patch line. Here they'll probably break the patching.

Comment (erykoff) [reply recipe/0001-loosen-pywin32-pin.patch]: Great point, I've moved the references.  But it turns out you can put anything you want at the top of a patch file, I believe it's ignored until it hits the `---` line.

Comment (ocefpaf) [reply recipe/0001-loosen-pywin32-pin.patch]: Ah. Yes! Still, we need to easily spot those during the next update to decide if we can drop them or not. So the best place is the recipe. Addition comments like, umerged, or merged and can be dropped next release also helps.

**Pull request merged**